### PR TITLE
Make extension optional in Windows

### LINKED ISSUE #1: Make extension optional on Windows?
Currently, you have to include the extension if you want to kill an application.

Eg. `fkill('notepad.exe')`

Would be nice if you just do `fkill('notepad')`.

If the user did `fkill('notepad')`, it could first try killing `notepad`, then `notepad.exe` if that didn't work.

Would love some thoughts on this from Windows users.

### PR DESCRIPTION
**Pull request by luizwbr**
Added '*' character to filter program's name, removing mandatory program extension.

Fixes #1

### CODE REVIEW

**Comment by MarkTiedemann** [comment]
This will kill potentially unrelated programs, e.g. `fkill note` would also kill `notepad`.

**Comment by luizwbr** [comment]
Yes, you are right. I will made some changes.

**Comment by sindresorhus** [comment]
Bump

**Comment by luizwbr** [comment]
Done!

File: index.js [hunk 1]
```diff
@@ -20,13 +20,25 @@ const missingBinaryError = async (command, arguments_) => {
 	}
 };
 
-const windowsKill = (input, options) => {
+const killWithTask = (input, options) => {
```
Comment (sindresorhus) [inline index.js]:
```suggestion
const killWithTaskkill = (input, options) => {
```

File: index.js [hunk 2]
```diff
@@ -20,13 +20,25 @@ const missingBinaryError = async (command, arguments_) => {
...
 	return taskkill(input, {
 		force: options.force,
 		tree: typeof options.tree === 'undefined' ? true : options.tree
 	});
 };
 
+const windowsKill = async (input, options) => {
+	try {
+		const firstTry = input;
```
Comment (sindresorhus) [inline index.js]: There variables are not useful.

File: index.js [hunk 3]
```diff
@@ -20,13 +20,25 @@ const missingBinaryError = async (command, arguments_) => {
...
+		return await killWithTask(firstTry, options);
```
Comment (sindresorhus) [inline index.js]:
```suggestion
		return await killWithTask(input, options);
```

File: index.js [hunk 4]
```diff
@@ -20,13 +20,25 @@ const missingBinaryError = async (command, arguments_) => {
...
+	} catch (error) {
+		if (typeof input === 'string') {
+			const secondTry = input.concat('.exe');
+			return killWithTask(secondTry, options);
```
Comment (sindresorhus) [inline index.js]:
```suggestion
			return killWithTask(input.concat('.exe'), options);
```

File: index.js [hunk 5]
```diff
@@ -20,13 +20,25 @@ const missingBinaryError = async (command, arguments_) => {
...
 		tree: typeof options.tree === 'undefined' ? true : options.tree
 	});
 };
 
+const windowsKill = async (input, options) => {
+	try {
+		const firstTry = input;
+		return await killWithTask(firstTry, options);
+	} catch (error) {
+		if (typeof input === 'string') {
```
Comment (sindresorhus) [inline index.js]: You are swallowing the error if it's not a string. When fixing this, also add a regression test.

File: test.js [hunk 6]
```diff
@@ -29,6 +29,15 @@ if (process.platform === 'win32') {
 		t.false(await processExists(pid));
 	});
 
+	test.serial('win default without extension', async t => {
```
Comment (sindresorhus) [inline test.js]:
```suggestion
	test.serial('window - supports extension-less process name', async t => {
```

**Comment by sindresorhus** [comment]
Support for this should be documented in readme and https://github.com/sindresorhus/fkill/blob/master/index.d.ts

**Comment by sindresorhus** [comment]
@luizwbr Still interested in finishing this?

**Pull request closed without merging**